packages/web3-core-requestmanager/src/index.js

```diff
@@ -123,11 +123,27 @@ RequestManager.prototype.setProvider = function (provider, net) {
             });
         });
 
-        // notify all subscriptions about the close condition
+        // notify all subscriptions about bad close conditions
         this.provider.on('close', function close(event) {
-            _this.subscriptions.forEach(function (subscription) {
-                subscription.callback(new Error('CONNECTION ERROR: The connection got closed with the close code `' + event.code + '` and the following reason string `' + event.reason + '`'));
-            });
+            // The node.net module emits ('close', isException)
+            const ipcCloseError = typeof event === 'boolean' && event;
+
+            // The WS package has a close event w/ codes & reasons
+            const hasErrorCode = ![1000, 1001].includes(event.code);
```

Comment:
``isCleanCloseEvent`` would be more explicit and I would prefer to move this out to a small private helper function on the ``RequestManager`` object. ``event.wasClean === false`` should be included in this helper function as well.

Reply:
``1000, 1001`` I've talked about to handle the error code ``1001`` differently because the end-point went away which is clearly an error from a client perspective. Only the error code ``1000`` does signal a clean close from the client. Do you think we can improve this?